tools/diff-processor/cmd/detect_missing_tests.go

```diff
@@ -0,0 +1,75 @@
+package cmd
+
+import (
+	newProvider "google/provider/new/google/provider"
+	oldProvider "google/provider/old/google/provider"
+
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/GoogleCloudPlatform/magic-modules/tools/diff-processor/detector"
+	"github.com/GoogleCloudPlatform/magic-modules/tools/diff-processor/diff"
+	"github.com/GoogleCloudPlatform/magic-modules/tools/diff-processor/reader"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+)
+
+const detectMissingTestsDesc = "Run the missing test detector using the given services directory"
+
+type detectMissingTestsOptions struct {
+	rootOptions *rootOptions
+}
+
+func newDetectMissingTestsCmd(rootOptions *rootOptions) *cobra.Command {
+	o := &detectMissingTestsOptions{
+		rootOptions: rootOptions,
+	}
+	return &cobra.Command{
+		Use:   "detect-missing-tests SERVICES_DIR",
+		Short: detectMissingTestsDesc,
+		Long:  detectMissingTestsDesc,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(c *cobra.Command, args []string) error {
+			return o.run(args)
+		},
+	}
+}
+
+func (o *detectMissingTestsOptions) run(args []string) error {
+	allTests, errs := reader.ReadAllTests(args[0])
+	for path, err := range errs {
+		glog.Infof("error reading path: %s, err: %v", path, err)
+	}
```

Comment:
```suggestion
	}
	if len(errs) > 0 {
		return fmt.Errorf("reading tests")
	}
```

Should we return an error here? So that the root command can halt execution if the tests can't be read. Perhaps that would address the issue of not running the unit tests on this branch / making sure we catch newly-introduced unparseable tests?

Reply:
Alternately we could return an error where the message includes the individual errors encountered (instead of logging them on separate lines)

Reply:
The missing test detector can still produce useful output even if the reader encounters errors.